configsuite/config.py

```diff
@@ -300,9 +318,9 @@ def _build_dict_snapshot(self, config, schema):
             ]
         )
 
-    def _build_snapshot(self, config, schema):
+    def _build_snapshot(self, config, schema, parent_type=None):
```

Comment:
Should not be necessary to touch the `snapshot` code. All the validation happens prior to this. The remaining changes in this PR should hit the `_build_initial_merged_config` as far as I can see.

Reply:
With the tests we have now it will be impossible to have the implementation in the `_build_snapshot` ;)

I barely started on trying to set this up in the `_build_initial_merged_config`, but encountered a few issues.. Didn't push the changes though, and I didn't get to the bottom of what was wrong before I had to go.. I'll be back on more on this!